apps/src/lib/node/ledger/shell/prepare_proposal/block_space_alloc.rs

```diff
@@ -0,0 +1,505 @@
+//! Primitives that facilitate keeping track of the number
+//! of bytes utilized by some Tendermint consensus round's proposal.
+//!
+//! This is important, because Tendermint places an upper bound
+//! on the size of a block, rejecting blocks whose size exceeds
+//! the limit stated in [`RequestPrepareProposal`].
+//!
+//! The code in this module doesn't perform any deserializing to
+//! verify if we are, in fact, allocating space for the correct
+//! kind of tx for the current [`BlockSpaceAllocator`] state. It
+//! is up to the user to dispatch the correct kind of tx into the
```

Comment:
The term `user` is confusing here. Maybe just say this logic is handled in the prepare proposal call.